base/mysqlxx/PoolWithFailover.cpp

```diff
@@ -136,6 +136,21 @@ PoolWithFailover::Entry PoolWithFailover::get()
 
                     if (!entry.isNull())
                     {
+                        /// For insert query chose only replicas with write access
+                        if (!check_write_access_to_table.empty())
+                        {
+                            auto query = entry->query("SELECT count(*) > 0 "
+                                                    "FROM information_schema.table_privileges "
+                                                    "WHERE privilege_type = 'INSERT' "
+                                                    "AND grantee = \"'" + pool->getUser() + "'@'%'\""
+                                                    "AND table_name = '" + check_write_access_to_table + "'");
+                            auto result{query.use()};
+                            auto row = result.fetch();
+                            if (!row || row[0].getUInt() == 0)
+                                continue;
```

Comment:
If no replicas have insert privilege, then `Connections to all replicas failed` will be thrown. Maybe we should throw exception with more specific message in this case? Something like "No replicas have insert privilege for ...".